### PR TITLE
Refactor PartitionTypeIndicator to use only Entities.

### PR DESCRIPTION
**Pull request by blattms**
Those have more information than EnitityRep which might be useful with LGRs/adaptivity.

This WIP as it is just there for demonstration.

### CODE REVIEW

**Comment by blattms** [comment]
jenkins build this please

File: opm/grid/cpgrid/PartitionTypeIndicator.cpp [hunk 1]
```diff
@@ -9,18 +9,18 @@ namespace Dune
...
-PartitionType PartitionTypeIndicator::getPartitionType(const EntityRep<0>& cell_entity) const
+PartitionType PartitionTypeIndicator::getPartitionType(const Entity<0>& cell_entity) const
 {
     if(cell_indicator_.size())
         return PartitionType(cell_indicator_[cell_entity.index()]);
     return InteriorEntity;
 }
 
-PartitionType PartitionTypeIndicator::getPartitionType(const EntityRep<1>& face_entity) const
+PartitionType PartitionTypeIndicator::getPartitionType(const Entity<1>& face_entity) const
```
Comment (aritorto) [inline opm/grid/cpgrid/PartitionTypeIndicator.cpp:19]: Do we support Entity<1> for CpGrid? Don't we always use EntityRep<1> instead?

File: opm/grid/cpgrid/CpGridData.cpp [hunk 2]
```diff
@@ -217,9 +217,9 @@ struct GetRowType<std::vector<E,A> >
 };
 
 PartitionType getPartitionType(const PartitionTypeIndicator& p, const EntityRep<1>& f,
-                               const CpGridData&)
+                               const CpGridData& grid)
 {
-    return p.getPartitionType(f);
+    return p.getPartitionType(Entity<3>(grid, f.index(), f.orientation()));
```
Comment (aritorto) [inline opm/grid/cpgrid/CpGridData.cpp:222]: Typo in the codim? for face, 1, for point 3. Same as below, just wondering if we also need some refactor in Entity to support a few methods there for Entity<1> for corner point grid

File: opm/grid/cpgrid/PartitionTypeIndicator.cpp (hunk 1 above)
Comment (blattms) [reply opm/grid/cpgrid/PartitionTypeIndicator.cpp:19]: I think we need some further refactoring to really drop that. When distributing the grid we ask for global ids on faces.

At least this change seems to work/compile.

Comment (aritorto) [reply opm/grid/cpgrid/PartitionTypeIndicator.cpp:19]: I see, I'm not familiar yet with global ids for faces (I did work on global ids for cells and points, and ignore faces so far). Step by step then!

File: opm/grid/cpgrid/CpGridData.cpp (hunk 2 above)
Comment (blattms) [reply opm/grid/cpgrid/CpGridData.cpp:222]: Good catch. I'll fix that.

**Comment by blattms** [comment]
I accidentally pushed this to OPM instead of my repo. I will deleted the branch here.

**Comment by blattms** [comment]
Update commit is c492f477a62065aefda98a493a971a23856d0eed

**Pull request closed without merging**